## **PRÁCTICA 1.1, 20/08/2026**
### **Victor Tavares Estrada. G.511**
### ***Tópicos de Big Data***

# **Actividad 1 - Identificación de la fuente**
## ***Propósito***
### Reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente. 

## **Cita de la fuente**
###  *UN, World Population Prospects (2024) – processed by Our World in Data. “Annual change in population – UN WPP” [dataset]. United Nations, “World Population Prospects”; United Nations, “World Population Prospects - Interim Update” [https://ourworldindata.org/grapher/annual-population-growth?v=1&csvType=full&useColumnShortNames=false].* 

## **Preguntas**
### Responder brevemente:
1. **¿Quién produjo originalmente los datos?** La Organización de las Naciones Unidas, en el informe *World Population Prospects (2024)*
2. **¿Qué organización los procesa para su utilización en Our World in Data?** Our World In Data *(OWID)*
3. **¿Qué representa la variable que analizarás?** El cambio anual en la población

# **Actividad 2 - Carga y exploración de datos**
## ***Propósito***
### Comprobar que el entorno de Python, Pandas y Jupyter funciona correctamente y realizar una primera exploración del dataset.

**2.1 Importar las bibliotecas**

In [67]:
import pandas as pd
import numpy as np

**2.2 Cargar el archivo CSV**

In [68]:
df = pd.read_csv('../data/raw/annual-population-growth/annual-population-growth.csv')

**2.3 Explorar el DataFrame**

In [69]:
# Primeras filas
df.head()


,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [70]:
# Dimensiones
df.shape

(38400, 5)

In [71]:
# Nombres de columnas
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [72]:
#Tipos de datos
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [73]:
# Estadísticas descriptivas
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [74]:
# Cantidad de valores faltantes
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [75]:
# Cantidad de Entidades (países)
df["Entity"].nunique()

256

## **Preguntas**
### Responder brevemente:
1. **¿Cuántas filas y columnas tiene el dataset?** 38,400 Filas, 5 Columnas
2. **¿Qué tipo de datos contiene la variable *Year*?** int64
3. **¿Existen valores faltantes?** Sí, en *"Code", "Annual change in population", y "Annual population change (Projected")*
4. **¿Cuántas entidades diferentes contiene el dataset?** 256 entidades (países)

# **Actividad 3 - Estadística Descriptiva**
## ***Propósito***
### Aplicar medidas estadísticas básicas a una variable poblacional. 

**Sobre la variable de cambio anual de población, calcular:**

In [76]:
año = 1951
df_filtrado = df[df["Year"] == año]

variable = df_filtrado['Annual change in population']

print(f"--- Estadísticas descriptivas para el año {año} ---")
print(f"Media: {variable.mean()}")
print(f"Mediana: {variable.median()}")
print(f"Varianza: {variable.var()}")
print(f"Desviación estándar: {variable.std()}")
print(f"Mínimo: {variable.min()}")
print(f"Máximo: {variable.max()}")

entidad_min = df_filtrado[df_filtrado['Annual change in population'] == variable.min()]['Entity'].values[0]
entidad_max = df_filtrado[df_filtrado['Annual change in population'] == variable.max()]['Entity'].values[0]
print(f"Entidad con el valor mínimo: {entidad_min}")
print(f"Entidad con el valor máximo: {entidad_max}")

--- Estadísticas descriptivas para el año 1951 ---
Media: 1123856.7109375
Mediana: 36355.5
Varianza: 23946448116977.24
Desviación estándar: 4893510.817090041
Mínimo: -533158.0
Máximo: 43834182.0
Entidad con el valor mínimo: North Korea
Entidad con el valor máximo: World


## **Preguntas**
### Responder brevemente:
1. **¿Cuál es el valor de la Media?** 1,123,856.71
2. **¿Cuál es el valor de la Mediana?** 36,355.6
3. **¿Cuál es el valor de la Desviación Estándar?** 4,893,510.81
4. **¿Qué entidad presenta el valor mínimo y cuál es el máximo?** *Mínimo:* North Korea, *Máximo:* Todo el mundo

# **Actividad 4 - Comprobación del entorno**
## ***Propósito***
### Comprobar que el notebook está utilizando correctamente las herramientas configuradas para la práctica. 

In [77]:
#Importar librería
import sys
import notebook

In [78]:
#Verificación de versiones
print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)
print("Jupyter Notebook:", notebook.__version__)

Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2
Jupyter Notebook: 7.6.2


# **Ejercicios extra de refuerzo**

In [79]:
#Un corchete = serie ( indice - valor) ( dataframe[year])
#Dos corchetes = formato dataframe ( tabla ) ( dataframe)
#SI se usa dos, te trae la información con formato dataframe ( tabla ) ( dataframe
df_year = df[["Year","Entity"]]
df_year.head()

,Year,Entity
0,1951,Afghanistan
1,1952,Afghanistan
2,1953,Afghanistan
3,1954,Afghanistan
4,1955,Afghanistan


In [80]:
#Filtro para las filas del año 2020
df_extraccion_filas = df[ df["Year"] == 2020]
df_extraccion_filas.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN
519,Algeria,DZA,2020,747543.0,NaN
669,American Samoa,ASM,2020,-451.0,NaN


In [81]:
#Varias condicionales para la extracción de filas
df_2020_mexico = df[(df["Entity"] == "Mexico") & (df["Year"] == 2020)]
df_2020_mexico.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [82]:
#Medidas con diferentes tipos de datos
df_2020_mexico.mean(numeric_only = True)

Year                                       2020.0
Annual change in population             1036075.0
Annual population change (Projected)          NaN
dtype: float64

In [83]:
#Localizar por el valor del índice LOC
df.loc[:, ["Entity", "Year"]]

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955
...,...,...
38395,Zimbabwe,2096
38396,Zimbabwe,2097
38397,Zimbabwe,2098
38398,Zimbabwe,2099


In [84]:
#Localizar por la posición del índice ILOC
df.iloc[30:35, 0:12]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
30,Afghanistan,AFG,1981,-1231727.0,NaN
31,Afghanistan,AFG,1982,-946206.0,NaN
32,Afghanistan,AFG,1983,-73395.0,NaN
33,Afghanistan,AFG,1984,272237.0,NaN
34,Afghanistan,AFG,1985,236633.0,NaN


In [85]:
#Localizar por el valor del índice LOC
df.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [86]:
df.loc[:, ['Entity', 'Year', 'Annual change in population']]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [87]:
#Combinar dos condiciones para filtrar filas y seleccionar columnas
df.loc[10000, ['Entity', 'Year', 'Annual change in population']]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [88]:
df.loc[(df['Entity'] == 'Mexico') & (df['Year'] == 2020)] 

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [89]:
df.iloc[0:5]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [90]:
df.iloc[:, [0,2,3]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [91]:
df.iloc[0:4, 0:3]

,Entity,Code,Year
0,Afghanistan,AFG,1951
1,Afghanistan,AFG,1952
2,Afghanistan,AFG,1953
3,Afghanistan,AFG,1954


In [92]:
#Filtrar por rango de años y por letra inicial del país
rango_años = (df['Year'] >= 2006) & (df['Year'] <= 2023)
letra_inicial_pais = df['Entity'].str.startswith('T')
df_filtrado = df.loc[rango_años & letra_inicial_pais]

variable = df_filtrado['Annual change in population']

print(f"Estadísticas del 2006 al 2023 para países que comienzan con la letra 'T':")
print(f"Media: {variable.mean()}")
print(f"Mediana: {variable.median()}")
print(f"Varianza: {variable.var()}")
print(f"Desviación estándar: {variable.std()}")
print(f"Mínimo: {variable.min()}")
print(f"Máximo: {variable.max()}")

entidad_min = df_filtrado[df_filtrado['Annual change in population'] == variable.min()]['Entity'].values[0]
entidad_max = df_filtrado[df_filtrado['Annual change in population'] == variable.max()]['Entity'].values[0]
print(f"Entidad con el valor mínimo: {entidad_min}")
print(f"Entidad con el valor máximo: {entidad_max}")

Estadísticas del 2006 al 2023 para países que comienzan con la letra 'T':
Media: 266065.42307692306
Mediana: 96120.5
Varianza: 225631829924.9919
Desviación estándar: 475007.18934032135
Mínimo: -138223.0
Máximo: 1978068.0
Entidad con el valor mínimo: Taiwan
Entidad con el valor máximo: Turkey


PRUEBA GITHUB